include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -222,11 +222,30 @@ class format_parse : public format_base
         return (std::find_if(begin_it, end_of_options_it,
             [&] (const std::string & v)
             {
-                size_t id_size{(prepend_dash(id)).size()};
-                if (v.size() < id_size)
-                    return false; // cannot be the correct identifier
+                std::string full_id = prepend_dash(id);
 
-                return v.substr(0, id_size) == prepend_dash(id); // check if prefix of v is the same
+                if constexpr (std::same_as<id_type, char>) // short id
+                {
+                    // check if prefix of v is the same since "-oValue" "-o=Value" "-o value" are all valid
+                    return v.substr(0, full_id.size()) == full_id;
+                }
+                else
+                {
+                    // only "--opt Value" or "--opt=Value" are valid
+                    if (v.substr(0, full_id.size()) == full_id)
+                    {
+                        if (v.size() == full_id.size())
+                            return true; // value was separated by space and is next in argv
+                        else if (v[full_id.size()] == '=')
```

Comment:
Is this save? 

What if my input in the arg list is `--validArg validValue --NoValue`. Would it access out of bounds?

Reply:
Yes!

But `if (v.substr(0, full_id.size()) == full_id)` should throw if `v.size() < full_id.size()`, because `substr` has a bounds check.
Can we construct an example where this might happen?
Is there a case where we might look up a very long ID in a argv that contains only short stuff?
It should throw if we look up `IAmAVeryLongID` in a argv of `["-not", "so", "--long"]`?

Reply:
I'll add a test!

Reply:
I added a test with a valid option `foo-bar` and a command line of `--foo value --foo-bar value`, which should trigger this case. It works as expected.
I looked up the documentation of `str.subtr(pos1, pos2)` again and if `pos2 > str.end()` then `str.subtr(pos1, str.end())` is returned and no exception is thrown.

Reply:
Yeah, I read it wrong, it throws if you do `str.substr(pos, count)`  and `pos` is out of bounds, if `pos+count` is out of bounds, it just returns the suffix starting at `pos` if `str`